src/lib/components/FileUpload/FileUpload.tsx

```diff
@@ -96,6 +101,7 @@ export const FileUpload: React.FC<FileUploadProps> = ({
               </span>
             ))}
           {files &&
+            !uploadingFiles &&
```

Comment:
This solution creates some problems. I'd expect files not yet uploaded would remain visible (and in the same order) even if some files have already been added to the 'uploadingFiles'. I'd suggest to maintain a single list for both idle and in progress states and assigning each file an individual status to track its progress.